### PR TITLE
Eslint fixes

### LINKED ISSUE #110: Review eslint warnings
We have thousands of warnings and are ignoring them. This means there is no point in having them.

Acceptance criteria:

- Look at .eslintrc.js
- Check all the rules with the word `warn` in them
- For each rule decide if it is worth having
  - If it's not useful, comment out the line
  - If it's useful, set it to error and fix all the issues

### PR DESCRIPTION
**Pull request by Beck-berry**
Fixes #110 
We updated the eslint settings in the `.eslintrc.js` file, so as there are no warnings, only errors or no notifications at all.
We fixed all errors afterwards so now no warnings nor errors should be given when running `yarn lint`. (Also no tests were harmed during the modifications.)

### CODE REVIEW

**Comment by leonardehrenfried** [comment]
This is good work, but I see a lot of these:

```
  lowEndLabel: undefined,
  highEndLabel: undefined,
  children: undefined
```
This quietens the linter but does it actually make the code better? If we do that we add cognitive load for the developer but we don't gain anything. Or have I overlooked something?

File: .eslintrc.js [hunk 1]
```diff
@@ -9,17 +9,20 @@ module.exports = {
   ],
   'rules': {
     'curly': ['error', 'all'],
-    'lines-between-class-members': 'warn',
-    'no-else-return': 'warn',
+    'lines-between-class-members': 'off',
+    'no-alert': 'off',
```
Comment (leonardehrenfried) [inline .eslintrc.js:13]: Do you want to allow calls to `alert()`?

Comment (Beck-berry) [reply .eslintrc.js:13]: There were 4 or 6 places when this alert occurred, and I was unsure what to exchange them to. If you say, we should not allow them, we should examine those more.

Comment (leonardehrenfried) [reply .eslintrc.js:13]: Yes, please re-check. `alert` opens an ugly modal window and usually you only want that if you have no other choice.

**Comment by Beck-berry** [comment]
There were several times I could not find a proper default value but eslint needed one, this is why undefined came so often. Should we then instead of making this eslint rule an error just turn it off? (Or perhaps keep only this one as a warning?)
I'll make the undefined values disappear, and only keep the useful modifications.

**Comment by leonardehrenfried** [comment]
> There were several times I could not find a proper default value but eslint needed one, this is why undefined came so often. Should we then instead of making this eslint rule an error just turn it off? (Or perhaps keep only this one as a warning?)
> I'll make the undefined values disappear, and only keep the useful modifications.

I feel that putting `undefined` everywhere just creates extra work for no gain. The goal is to make the code better not to make the linter quiet. :)

In such a case I would just set it to off but if you want to keep the warning, then that's also fine.

File: .eslintrc.js (hunk 1 above)
Comment (Beck-berry) [reply .eslintrc.js:13]: As I found out, these alerts come in the special admin page (no user should see them ever), so I disabled eslint in these lines.

**Comment by leonardehrenfried** [comment]
Ok, so you would like to keep the warning for the default prop value?

**Comment by Beck-berry** [comment]
I'm not sure about this. I think that giving default values whenever it's possible is a good point but if we keep this as a warning, I don't think we would ever scroll through the list to find a newly created one, so... Only if we could've make these existing ones disappear would it make sense for the future to look at this rule as a serious one.
All in all, might be better to turn it off.

**Pull request merged**